Detectors/GlobalTrackingWorkflow/tofworkflow/src/TOFCalibWriterSpec.cxx

```diff
@@ -0,0 +1,125 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @brief  Processor spec for a ROOT file writer for TOF digits
+
+#include "TOFWorkflow/TOFCalibWriterSpec.h"
+#include "Framework/CallbackService.h"
+#include "Framework/ControlService.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/DataRefUtils.h"
+#include "Framework/Lifetime.h"
+#include "Framework/Task.h"
+#include "Framework/SerializationMethods.h"
+#include "Headers/DataHeader.h"
+#include <SimulationDataFormat/MCCompLabel.h>
+#include <SimulationDataFormat/MCTruthContainer.h>
+#include "TTree.h"
+#include "TBranch.h"
+#include "TFile.h"
+#include "ReconstructionDataFormats/MatchInfoTOF.h"
+#include "ReconstructionDataFormats/CalibInfoTOF.h"
+#include "DataFormatsTOF/Cluster.h"
+#include <memory> // for make_shared, make_unique, unique_ptr
+#include <vector>
+
+using namespace o2::framework;
+using SubSpecificationType = o2::framework::DataAllocator::SubSpecificationType;
+
+namespace o2
+{
+namespace tof
+{
+
+  using evIdx = o2::dataformats::EvIndex<int, int>;
+  using OutputType = std::vector<o2::dataformats::CalibInfoTOF>;
+
+template <typename T>
+TBranch* getOrMakeBranch(TTree& tree, std::string brname, T* ptr)
+{
+  if (auto br = tree.GetBranch(brname.c_str())) {
+    br->SetAddress(static_cast<void*>(&ptr));
+    return br;
+  }
+  // otherwise make it
+  return tree.Branch(brname.c_str(), ptr);
+}
+
+/// create the processor spec
+/// describing a processor receiving calib info for TOF writing them to file
+/// TODO: make this processor generic and reusable!!
+DataProcessorSpec getTOFCalibWriterSpec()
```

Comment:
Since all DPL devices are now using Task - driven methods, can this also be converted to task? It will be easier to manage.